batch/plugins/org.jboss.tools.batch.ui/src/org/jboss/tools/batch/ui/hyperlink/BatchPropertyHyperlinkDetector.java

```diff
@@ -0,0 +1,150 @@
+import org.eclipse.jdt.internal.ui.javaeditor.EditorUtility;
+import org.eclipse.jdt.internal.ui.javaeditor.JavaEditor;
+import org.eclipse.jdt.internal.ui.text.JavaWordFinder;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.ITextViewer;
+import org.eclipse.jface.text.hyperlink.AbstractHyperlinkDetector;
+import org.eclipse.jface.text.hyperlink.IHyperlink;
+import org.eclipse.ui.IFileEditorInput;
+import org.eclipse.ui.texteditor.ITextEditor;
+import org.jboss.tools.batch.core.IBatchArtifact;
+import org.jboss.tools.batch.core.IBatchProject;
+import org.jboss.tools.batch.core.IBatchProperty;
+import org.jboss.tools.batch.internal.core.impl.BatchProject;
+import org.jboss.tools.batch.internal.core.impl.BatchProjectFactory;
+import org.jboss.tools.batch.internal.core.impl.BatchUtil;
+import org.jboss.tools.batch.ui.BatchUIPlugin;
+import org.jboss.tools.common.text.ext.util.Utils;
+import org.w3c.dom.Attr;
+
+public class BatchPropertyHyperlinkDetector extends AbstractHyperlinkDetector {
+	protected IFile file;
+	
+	@Override
+	public IHyperlink[] detectHyperlinks(ITextViewer textViewer, IRegion region,
+			boolean canShowMultipleHyperlinks) {
+		List<IHyperlink> links = new ArrayList<IHyperlink>();
+
+		ITextEditor textEditor= (ITextEditor)getAdapter(ITextEditor.class);
+		if (region == null || !(textEditor instanceof JavaEditor))
+			return null;
+		
+		if(textEditor.getEditorInput() instanceof IFileEditorInput){
+			file = ((IFileEditorInput)textEditor.getEditorInput()).getFile();
+		}
+		
+		int offset= region.getOffset();
+		
+		ITypeRoot input= EditorUtility.getEditorInputJavaElement(textEditor, true);
+		if (input == null)
+			return null;
+
+		IDocument document= textEditor.getDocumentProvider().getDocument(textEditor.getEditorInput());
+		IRegion wordRegion= JavaWordFinder.findWord(document, offset);
+		if (wordRegion == null)
+			return null;
+		
+		IProject project = null;
+		
+		project = input.getJavaProject().getProject();
+		
+		if(project == null)
+			return null;
+		
+		BatchProject batchProject = (BatchProject) BatchProjectFactory.getBatchProjectWithProgress(project);
+
+		if (batchProject == null) {
+			return null;
+		}
+		
+		IJavaElement[] elements = null;
+		
+		try {
+			elements = input.codeSelect(wordRegion.getOffset(), wordRegion.getLength());
+			
+			if (elements == null) 
+				return null;
+			
+			if(elements.length != 1)
+				return null;
+			
+			ArrayList<IHyperlink> hyperlinks = new ArrayList<IHyperlink>();
+			
+			IField field = null;
+			
+			if(elements[0] instanceof IType){
+				elements[0] = input.getElementAt(wordRegion.getOffset());
+				
+				if(elements[0] == null)
+					return null;
+				
+				if(elements[0] instanceof IField){
+					field = (IField) elements[0];
+				}
+			}else if(elements[0] instanceof IField){
+				field = (IField) elements[0];
+			}
+			
```

Comment:
Check if(elements[0] instanceof IField) is done twice. Code can be rewritten as:

if(elements[0] instanceof IType) {
    elements[0] = input.getElementAt(wordRegion.getOffset());
}
if(elements[0] instanceof IField) {
    field = (IField) elements[0];
}